Doc/library/stdtypes.rst

```diff
@@ -2841,6 +2841,17 @@ objects.
          optional *sep* and *bytes_per_sep* parameters to insert separators
          between bytes in the hex output.
 
+   .. method:: resize(size)
+
+      Resize the :class:`bytearray` using :ref:`Memory Management <memory>` APIs
+      to contain *size* bytes. Data from the existing :class:`bytearray` will be
+      kept up to *size*. If :class:`bytearray` needs to grow, all new bytes will
+      be zeroed / set to the NULL byte (``b"\0"``). One byte beyond *size* will
+      always be set to NULL to ensure the :class:`bytearray` is always NULL
+      terminated regardless of contents.
```

Comment:
> using :ref:`Memory Management <memory>` APIs

We don't document the implementation usually.

> Data from the existing :class:`bytearray` will be kept up to *size*.

Is it worth it to document it?

> One byte beyond *size* will always be set to NULL to ensure the :class:`bytearray` is always NULL terminated regardless of contents.

That's only true in C, it's not visible in Python. And it's an implementation details. I don't think that it should be documented (in the Python documentation).

I suggest:

```suggestion
      Resize the :class:`bytearray` to contain *size* bytes.
      If :class:`bytearray` needs to grow, all new bytes will be set to null bytes.
```